rollup/app/lwc/rollupForceRecalculation/rollupForceRecalculation.html

```diff
@@ -1,186 +1,188 @@
+              <lightning-input
+                class="slds-col slds-form-element slds-form-element_horizontal"
+                data-id="GrandparentRelationshipFieldPath__c"
+                label="Grandparent Relationship Field Path (optional)"
+                name="GrandparentRelationshipFieldPath__c"
+                oncommit={handleChange}
+                type="text"
+              >
+              </lightning-input>
+              <lightning-input
+                class="slds-col slds-form-element slds-form-element_horizontal"
+                data-id="OneToManyGrandparentFields__c"
+                label="One To Many Grandparent Fields (optional)"
+                name="OneToManyGrandparentFields__c"
+                oncommit={handleChange}
+                type="text"
+              >
+              </lightning-input>
+              <lightning-helptext
+                content="Use ObjectApiName.FieldName (or ObjectApiName__c.CustomField__c for custom objects and fields). Can be a comma-separated list for multiple junction object hops"
+              ></lightning-helptext>
+            </div>
+            <div class="slds-grid">
+              <lightning-input
+                class="slds-col slds-form-element slds-form-element_horizontal slds-m-vertical_small"
+                data-id="SplitConcatDelimiterOnCalcItem__c"
+                label="Split Concat Delimiter On Child Object, Too? (Optional)"
+                name="SplitConcatDelimiterOnCalcItem__c"
+                oncommit={handleChange}
+                type="checkbox"
+              >
+              </lightning-input>
+              <lightning-helptext content="Only include the split option for CONCAT_DISTINCT rollups"></lightning-helptext>
+            </div>
+            <div class="slds-grid">
+              <lightning-textarea
+                class="slds-col slds-form-element slds-form-element_horizontal"
+                label="SOQL Where Clause To Exclude Calc Items (Optional)"
+                name="CalcItemWhereClause__c"
+                onchange={handleChange}
+              ></lightning-textarea>
+              <lightning-helptext content="If including a SOQL where clause, do not start with 'WHERE'- this is added automatically"></lightning-helptext>
+            </div>
+          </template>
+          <lightning-button class="slds-col slds-m-top_small" label="Start rollup!" onclick={handleSubmit} variant="brand"></lightning-button>
+        </div>
+      </lightning-layout>
+      <template if:true={isRollingUp}>
+        <div>Rollup processing ...</div>
+        <lightning-spinner alternative-text="Please wait while rollup is processed" title="Rolling up ...."></lightning-spinner>
+      </template>
+    </section>
     <template if:true={rollupStatus}>
-      <div class="slds-m-top_small">Rollup job status: <b>{rollupStatus}</b></div>
+      <div class="slds-m-top_small">Rollup job status: <b>{rollupStatus} for job: {jobIdToDisplay}</b></div>
```

Comment:
One thing I came across when testing this out in a scratch org - I deployed the source & immediately went to this LWC (I didn't configure `RollupSettings__c` or anything like that). When I clicked 'Start rollup!', it says `Rollup job status: No process Id for job:` (screenshot below) - it took me a minute of tinkering to realize that I needed to create a record in `RollupSettings__c`. Once I created the record, then the 'Start rollup' button worked fine (I didn't even have to refresh the page, I just clicked the button again & it worked). It seems to be because `RollupSettings__c.getInstance().IsEnabled__c` returns `false` when there aren't any records created, even though the default value for `IsEnabled__c` is set to `true`.

Should this LWC care if the custom setting is enabled for the current user? If so, I think having a more descriptive error message should be displayed.

![image](https://user-images.githubusercontent.com/1267157/198181104-2411eb35-e470-478b-9f33-1263f4bac785.png)

Reply:
Ah that’s interesting, I thought I had made a change a few versions back to correctly pass back a string telling the user the custom setting needed to be enabled. Let me investigate this, I may make this change in the next version (to either pass back a more descriptive string or to bypass the custom setting for full recalcs)

Reply:
Yeah, I think it can wait, it doesn't seem like a big deal for most orgs, it's more of an edge case

Reply:
Oh yeah, forgot to follow up here since we chatted about it on the side - but for posterity, I fixed this too! (And, with your feedback, we made it strongly typed field names so that even in the namespaced version if the custom setting/rollup control field means a rollup doesn't process, the error message uses the namespace-appropriate version of the field)